src/basic/hashmap.c

```diff
@@ -2130,8 +2133,7 @@ int _hashmap_dump_sorted(HashmapBase *h, void ***ret, size_t *ret_n) {
 
         if (_hashmap_size(h) == 0) {
                 *ret = NULL;
-                if (ret_n)
-                        *ret_n = 0;
+                *ret_n = 0;
```

Comment:
if ret_n is not optional, you should add `assert(ret_n)` up top, next to the `assert(ret)`